src/Filters/FiltersScope.php

```diff
@@ -3,17 +3,43 @@
 namespace Spatie\QueryBuilder\Filters;
 
 use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Arr;
 use Illuminate\Support\Str;
+use ReflectionObject;
 
 class FiltersScope implements Filter
 {
-    public function __invoke(Builder $query, $values, string $property)
+    public function __invoke(Builder $query, $values, string $property): Builder
     {
         $scope = Str::camel($property);
 
         $values = Arr::wrap($values);
+        $values = $this->resolveTypeHintedParameters($query, $values, $scope);
 
-        $query->$scope(...array_values($values));
+        return $query->$scope(...$values);
+    }
+
+    protected function resolveTypeHintedParameters(Builder $query, $values, string $scope): array
```

Comment:
Can we rename this to `resolveParameters`? 90% of the time they're just gonna be regular params instead of type hinted parameters

Reply:
Same for the `$scopeParameters` variable :)